src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java

```diff
@@ -231,6 +221,28 @@ protected void updateStockWardQuantity(MovementWard movement) throws OHServiceEx
 		}
 	}
 
+	/**
+	 * Creates a new {@link MedicalWard} from the specified {@link MovementWard}.
+	 * @param movement the movement ward.
+	 * @param qty the quantity.
+	 * @return the created medical ward.
+	 */
+	private MedicalWard createMedicalWard(MovementWard movement, float qty) {
```

Comment:
Yes there is a lot of common code but it is very straight forward and clear as to what is being set when.   I find I have to really study this new method to understand what is happening and when.   I don't think this is useful IMO.